packages/disclosure/src/index.js

```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import Component from "@reach/component-component";
+import PropTypes from "prop-types";
+
+const Disclosure = ({ buttonLabel, children }) => (
+  <Component getInitialState={() => ({ isExpanded: false })}>
+    {({ state, setState }) => (
+      <React.Fragment>
+        <button
+          aria-expanded={state.isExpanded}
+          data-reach-disclosure-button
+          onClick={() =>
+            setState(({ isExpanded }) => ({ isExpanded: !isExpanded }))
+          }
+        >
+          <span aria-hidden={true}>{state.isExpanded ? "▼" : "►"}</span>
+          {buttonLabel}
+        </button>
+        {state.isExpanded && (
+          <div data-reach-disclosure-content>{children}</div>
+        )}
+      </React.Fragment>
+    )}
+  </Component>
+);
+
+Disclosure.propTypes = {
+  buttonLabel: PropTypes.string.isRequired
```

Comment:
I'm not aware of any [component (API) design guidelines] for this library, but the trend does seem to teeter more on the pattern of compound components. With the [button content] being a prop, that's a string, the consumer of this component would be limited with how they want to render the button; e.g. the following wouldn't be possible:

```jsx
<Disclosure
  buttonLabel={
    <React.Fragment>
      More <strong className="flashy">info</strong>
    </React.Fragment>
  }
>
  <p>content</p>
</Disclosure>
```

At the very least, I'd consider making the prop-type `node`; but perhaps using the context API and compound components pattern, you could mimic the API of the native HTML [`details`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details) element:

```jsx
import { Disclosure, DisclosureSummary, DisclosureContent } from '@reach/Disclosure';

// ...

<Disclosure>
  <DisclosureSummary>
    More <strong className="flashy">info</strong>
  </DisclosureSummary>
  <DisclosureContent>content</DisclosureContent>
</Disclosure>
```

Reply:
Hey @VinceMalone,

I think you've made a good point and I would be more than happy to update the API. I think a render prop could work quite well in this use case for the `buttonContent`. Would be interesting to know what @ryanflorence thinks :+1: